### PR TITLE
Add Reserve Form With Connection To Reserve API #1 #5

### LINKED ISSUE #1: Reserve DOI UI forms

### LINKED ISSUE #5: Connect UI (reserve) to API
Connect the UI to a API deployed on dev.

### PR DESCRIPTION
**Pull request by eddiesarevalo**
The form for entering reserve API parameters including a parsed excel sheet has been added. The form includes a submit button that will send the parameters and parsed excel sheet to the reserve API.

Resolves #1  
Resolves #5

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,72 @@
 # pds-doi-ui
 web UI for pds-doi-service
+
+This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
+
+## Available Scripts
+
+In the project directory, you can run:
+
+### `npm start`
```
Comment (tloubrieu-jpl) [inline README.md:10]: Can you make the README a bit more specific.

Especially, we are missing the `npm install` as a first step I think.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
+{
+  "name": "doi-ui",
```
Comment (tloubrieu-jpl) [inline package.json:2]: The (new) standard is to name the packages with pds- prefix, but have the repo name without the pds- prefix (the opposite of waht we usually do).
For now can you just rename the package pds-doi-ui.

File: src/components/ReserveExcelContent.jsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+    const generateExcelContent = () => {
+        let content = '';
+
+        if(excelContent && excelContent[0]){
+            return (
+                <div>
+                    <br></br>
+                    <Container>
+                        <TableContainer component={Paper}>
```
Comment (tloubrieu-jpl) [inline src/components/ReserveExcelContent.jsx:25]: I was hoping a component could do everything, read the excel, generate the editable spreadsheet table and save it.

But that is fine for now.

File: src/sagas/index.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+function* sendReserveContent(action){
+    const reserveAction = action.payload.action;
+    const submitter = action.payload.submitter;
+    const node = action.payload.node;
+    const url = action.payload.url;
+
+    let json = action.payload.excelContent;
+    json = JSON.stringify(json);
+
+    let endpoint = 'http://localhost:8085/PDS_APIs/pds_doi_api/0.1/dois';
```
Comment (tloubrieu-jpl) [inline src/sagas/index.js:12]: This would need to be in a configuration file somewhere

**Review by tloubrieu-jpl — CHANGES_REQUESTED** [review]
Thanks Eddie.
I was not able to make reserve button work.

I have also some comments on the UI, see 
Screen Shot 2020-11-19 at 10 06 14 AM

File: README.md (hunk 1 above)
Comment (eddiesarevalo) [reply README.md:10]: Sure thing! This readme was auto generated by react but I can go through and make it more specific.

File: package.json (hunk 2 above)
Comment (eddiesarevalo) [reply package.json:2]: Got it, I'll make this change.

File: src/components/ReserveExcelContent.jsx (hunk 3 above)
Comment (eddiesarevalo) [reply src/components/ReserveExcelContent.jsx:25]: I can add that in if you want. We should create a new issue for this though.

File: src/sagas/index.js (hunk 4 above)
Comment (eddiesarevalo) [reply src/sagas/index.js:12]: I'll move this out to a config.

**Comment by eddiesarevalo** [comment]
> Thanks Eddie.
> I was not able to make reserve button work.
> 
> I have also some comments on the UI, see
> Screen Shot 2020-11-19 at 10 06 14 AM

Thank you for your feedback, I have only been doing minimal styling so far so I can take it into account. I have just been working on functionality not really the styling. I was thinking of doing the styling on a different issue as I wanted to use Mikes styling from the DOI and to make the forms look better. This takes some time though. I am not sure if we should include that in this pull request or after the app functionality is complete as it is simpler nearer the end.

File: src/components/ReserveExcelContent.jsx (hunk 3 above)
Comment (tloubrieu-jpl) [reply src/components/ReserveExcelContent.jsx:25]: I will create the issue in the backlog

Comment (tloubrieu-jpl) [reply src/components/ReserveExcelContent.jsx:25]: https://github.com/NASA-PDS/pds-doi-ui/issues/7

**Comment by tloubrieu-jpl** [comment]
Thanks @eddiesarevalo for your feedback. The styling is not critical at all but the action menu should be removed and the reserve button should work (it did not in my deployment).

Thanks

**Comment by eddiesarevalo** [comment]
@tloubrieu-jpl I have made the changes except for the editable table and the styling which will come later. When you attempted to run this locally were there any errors thrown on the developer console or network tab? 
For the call to work a submitter email and node must be entered. Also an excel sheet must be uploaded.

**Review by tloubrieu-jpl — APPROVED** [review]

**Pull request merged**